src/SFA.DAS.PAS.Account.Api/Orchestrator/EmailOrchestrator.cs

```diff
@@ -15,65 +17,111 @@ public class EmailOrchestrator
+                    if (!recipients.Any())
+                    {
+                        _logger.Warn("All recipients explicitly requested for email are absent from Provider IDAMS - email message will be suppressed");
+                        return;
+                    }
+                }
             }
 
             if (!recipients.Any())
             {
-                recipients = await GetIdamsRecipients(ukprn);
+                recipients = idamsUsers.Any() ? idamsUsers : idamsSuperUsers;
             }
 
-            if(!recipients.Any())
+            if (!recipients.Any())
             {
-                recipients = await GetAccountUserRecipients(ukprn);
+                recipients = accountUsers.Select(x=>x.EmailAddress).ToList();
             }
 
-            var commands = recipients.Select(x => new SendNotificationCommand{ Email = CreateEmailForRecipient(x, message) });
+            var optedOutList = accountUsers.Where(x => !x.ReceiveNotifications).Select(x => x.EmailAddress).ToList();
+
+            var finalRecipients = recipients.Where(x =>
+                !optedOutList.Any(y => x.Equals(y, StringComparison.CurrentCultureIgnoreCase)))
+                .ToList();
+
+            var commands = finalRecipients
+                .Select(x => new SendNotificationCommand{ Email = CreateEmailForRecipient(x, message) });
             await Task.WhenAll(commands.Select(x => _mediator.Send(x)));
-        }
 
-        private async Task<List<string>> GetAccountUserRecipients(long ukprn)
-        {
-            var accountUsers = await _accountOrchestrator.GetAccountUsers(ukprn);
-            return accountUsers.Where(x => x.ReceiveNotifications).Select(x => x.EmailAddress).ToList();
+            _logger.Info($"Sent email to {finalRecipients.Count} recipients for ukprn: {providerId}", providerId);
         }
 
-        private async Task<List<string>> GetIdamsRecipients(long ukprn)
+        private async Task RemoveAccountUsersNotInIdams(List<User> accountUsers, List<string> allIdamsUsers)
         {
-            var recipients = await _idamsEmailServiceWrapper.GetEmailsAsync(ukprn);
-            if (recipients.Any())
+            var removedUsers = new List<User>();
+
+            foreach (var accountUser in accountUsers)
             {
-                return recipients;
+                if (!allIdamsUsers.Contains(accountUser.EmailAddress, StringComparer.CurrentCultureIgnoreCase))
+                {
+                    removedUsers.Add(accountUser);
+                }
             }
 
-            recipients = await _idamsEmailServiceWrapper.GetSuperUserEmailsAsync(ukprn);
-            if (recipients.Any())
+            foreach (var user in removedUsers)
             {
-                return recipients;
+                _logger.Info($"Account User {user.UserRef} not found in IDAMS and will be marked as deleted");
+                await _mediator.Send(new DeleteRegisteredUserCommand { UserRef = user.UserRef });
```

Comment:
Considering we have a list of users to remove would it be better to remove them as a single call

Reply:
I think removal from IDAMS will be relatively uncommon (maybe not first time this code runs, but from then on).